### DISCUSSION TITLE
Maximum number of connections from user+IP exceeded

### DISCUSSION THREAD

**Discussion by kshitijkurandwad211** [Mailu/Mailu#3659] [User support]
I'm getting IMAP limit with number of connections on a particular IP as 20. I want to increase it to at least 50. Any idea?
Saw few other discussions and create an override file in /overrides/dovecot/ with name dovecot.conf, with following content 
```
protocol imap {
   mail_max_userip_connections = 100
}
```

But still when I look at main conf file after restarting imap container it doesn't look like value is overwritten. Please help 🙏

**Comment by nextgens** [comment]
``mail_max_userip_connections`` is the maximum number of connections **for a given user** per IP. Do you really have a user who needs more than 20 connections from the same IP address?

Aren't you trying to tackle the problem at the wrong end here?

**Reply by nextgens** [reply]
Share the logs please

**Reply by kshitijkurandwad211** [reply]
Yes we have a setup where there are 20 folders in which mail goes after filtering. These folders are connected from a node js application. So basically there are 20 imap connections one for each folder. Which do processing after reading mail.

**Reply by kshitijkurandwad211** [reply]
Logs after restarting the server?

**Reply by nextgens** [reply]
> Yes we have a setup where there are 20 folders in which mail goes after filtering. These folders are connected from a node js application. So basically there are 20 imap connections one for each folder. Which do processing after reading mail.

This is definitely the wrong tool for the wrong job. Look into https://rspamd.com/doc/modules/metadata_exporter.html if you control the application; you can get rspamd to POST to your app when it receives something.

**Reply by nextgens** [reply]
Adding to the above, you could also use a sieve rule to do the same; It may be even simpler.

**Reply by kshitijkurandwad211** [reply]
Yes, using sieve they are filtered to their respective folders. For example "Subject:"Bug"" goes to Bug folder. This new mail is read using imap connection on  of application parse the mail and apply necessary logics needed for Bug mail something like creating a ticket in our jira system.

**Reply by kshitijkurandwad211** [reply]
> > Yes we have a setup where there are 20 folders in which mail goes after filtering. These folders are connected from a node js application. So basically there are 20 imap connections one for each folder. Which do processing after reading mail.
> 
> This is definitely the wrong tool for the wrong job. Look into https://rspamd.com/doc/modules/metadata_exporter.html if you control the application; you can get rspamd to POST to your app when it receives something.

Is it possible that this happens after mail filtering? Is yes please help me with some docs or guidance

**Reply by nextgens** [reply]
All of the above would happen after spam filtering.

My recommendation is to use sieve if you have a single mailbox. See
https://datatracker.ietf.org/doc/html/rfc5435

**Reply by kshitijkurandwad211** [reply]
I already have sieve script that filters email into their respective folders when it arrives. I'm talking about post moving it to their respective folders, I need to process their emails based on content in my node js application

**Reply by nextgens** [reply]
You can get the sieve scripts to POST the content of the emails to your application.

**Reply by kshitijkurandwad211** [reply]
Let me make it more clear for you. I have a mail server with one main inbox, with mail id: support@<custom>.com
This mail box has structure:
 INBOX
 SPAM
 DELETED
 ..
 BUGS
 FEEDBACK
(.... 10 others)

I have sieve script that filters mail to BUGS, FEEDBACK etc. Rest of mails are left in inbox(these are not needed). These custom folders have IMAP connections, which further get the mail( onMail event is attached in this). Reads it, create tickets based on folder and email content.
I don't think there is a way where I can run script like when a new mail arrives at BUGS, do a post call to https://<server-url>. Also these mail filtering to folder is done using Roundcube. 
So when our folders are more than 20 it is causing an issues that same ip is trying to create more than 20 connections as there are more than 20 folders.
If there is way to make that post call please guide me
